src/lib/paymentOptions/NimiqPaymentOptions.ts

```diff
@@ -14,23 +15,31 @@ export interface NimiqSpecifics {
     recipientType?: Nimiq.Account.Type;
 }
 
-export type ParsedNimiqSpecifics = Omit<NimiqSpecifics, 'sender' | 'recipient' | 'fee' | 'feePerByte'>
+export type ParsedNimiqSpecifics = Omit<NimiqSpecifics, 'sender' | 'recipient' | 'fee' | 'feePerByte' | 'extraData'>
     & Required<Pick<NimiqSpecifics, 'forceSender' | 'flags' | 'validityDuration'>> & {
     sender?: Nimiq.Address,
     recipient?: Nimiq.Address,
-    fee: number,
-    feePerByte: number,
+    fee?: number,
+    feePerByte?: number,
+    extraData?: Uint8Array,
 };
 
 export type NimiqDirectPaymentOptions = PaymentOptions<Currency.NIM, PaymentType.DIRECT>;
 
 export class ParsedNimiqDirectPaymentOptions extends ParsedPaymentOptions<Currency.NIM, PaymentType.DIRECT> {
     public amount: number;
-    private extraData: Uint8Array;
 
-    public constructor(options: NimiqDirectPaymentOptions, extraData: Uint8Array) {
+    public constructor(options: NimiqDirectPaymentOptions, enforceDefaultValues = true) {
         super(options);
-        this.extraData = extraData;
+
+        if (options.protocolSpecific.extraData !== undefined && typeof options.protocolSpecific.extraData !== 'string'
```

Comment:
I think the first check can be simply a truthy check. Or do you want to allow an empty string?

Reply:
It should not throw for an empty string. The prior default was `new Uint8Array(0)`, which kind of is similar to `''`.